test/controllers/admin/subdomains_controller_test.rb

```diff
@@ -120,6 +120,34 @@ class Admin::SubdomainsControllerTest < ActionDispatch::IntegrationTest
     end
   end
 
+  test 'allows #create if global admin & invites them to the subdomain with permission to manage everything' do
+    sign_in(@user)
+    name = 'foobar'
+    assert_difference "Subdomain.all.size", +1 do
+      payload = {
+        subdomain: {
+          name: name
+        }
+      }
+      perform_enqueued_jobs do
+        assert_changes "Devise::Mailer.deliveries.size" do
+          assert_difference "Subdomain.count", +1 do
+            post admin_subdomains_url, params: payload
+            Apartment::Tenant.switch name do
+              assert User.first
```

Comment:
make sure to query the database only once by referencing user as `user = User.first`